sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/EmptyCollectionsShouldNotBeEnumerated.cs

```diff
@@ -244,21 +268,21 @@ private ProgramState ProcessElementAccess(ProgramState programState, ElementAcce
                 }
 
                 return constraint != null
-                    ? symbolicValue.SetConstraint(constraint, programState)
-                    : programState;
+                    ? symbolicValue.SetConstraint(constraint, newProgramState)
+                    : newProgramState;
             }
 
             private static bool IsIgnoredMethod(IMethodSymbol methodSymbol)

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Consider using more general type 'Microsoft.CodeAnalysis.ISymbol' instead of 'Microsoft.CodeAnalysis.IMethodSymbol'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS3242)